client/src/components/BlogPostForm.tsx

```diff
@@ -0,0 +1,194 @@
+import { AiOutlineLeft } from 'react-icons/ai';
+import { Link } from 'react-router-dom';
+import { FaLocationDot } from 'react-icons/fa6';
+import { Entry, Photo } from '../lib/types';
+import { FormEvent, useState } from 'react';
+import UnsplashGallery from './UnsplashGallery';
+
+type props = {
+  entry: Entry | undefined;
+};
+
+export default function BlogPostForm({ entry }: props) {
+  const [title, setTitle] = useState(entry?.title ?? '');
+  const [subtitle, setSubtitle] = useState(entry?.subtitle ?? '');
+  const [location, setLocation] = useState(entry?.location ?? '');
+  const [photoInfo, setPhotoInfo] = useState(entry?.photoInfo ?? undefined);
+  const [body, setBody] = useState(entry?.body);
+  const [error, setError] = useState<unknown>();
+  const [search, setSearch] = useState('');
+  const [photos, setPhotos] = useState<Array<Photo>>();
+  const [missingPhoto, setMissingPhoto] = useState(entry ? false : undefined);
+
+  async function handleSubmit(event: FormEvent) {
+    event.preventDefault();
+    if (!photoInfo) {
+      setMissingPhoto(true);
+      return;
+    }
+    try {
+      if (!title || !subtitle || !location || !body) {
+        throw new Error('Error: 400 invalid request');
+      }
+      const {
+        urls: { regular },
+        user: {
+          name,
+          links: { self },
+        },
+        alt_description,
+      } = photoInfo;
+      const entryInput: Entry = {
+        title,
+        subtitle,
+        location,
+        photoInfo,
+        body,
+        url: regular,
+        author: name,
+        authorURL: self,
+        alt: alt_description,
+      };
+      const req = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(entryInput),
+      };
+      const res = await fetch('/api/entries', req);
+      if (!res.ok) {
+        throw new Error(`fetch Error ${res.status}`);
+      }
```

Comment:
I would pull this entire block of code into a separate function, like `createEntry`.